wit/couchbase/subdocument.wit

```diff
@@ -0,0 +1,307 @@
+package wasmcloud:couchbase@0.1.0-draft;
+
+/// Perform sub-document lookup operations on Documents stored in a Couchbase cluster
+///
+/// Reference: https://docs.couchbase.com/go-sdk/current/howtos/subdocument-operations.html#retrieving
+interface subdocument-lookup {
+  use types.{
+    document-id, request-span, document, subdocument-path, document-error, retry-strategy, mutation-metadata
+  };
+
+  /// Options for performing a lookup
+  record lookup-get-options {
+    /// Whether the lookup refers to an extended attribute, rather than the document itself
+    xattr: option<bool>,
+  }
+
+  type lookup-exists-options = lookup-get-options;
+  type lookup-count-options = lookup-get-options;
+
+  /// Types of different lookup operations that can be specified
+  variant lookup-operation {
+    /// Whether the particular path exists
+    exists(tuple<subdocument-path, lookup-exists-options>),
+    /// Whether the particular path exists
+    get(tuple<subdocument-path, lookup-get-options>),
+    /// Whether the particular path exists
+    count(tuple<subdocument-path, lookup-count-options>),
+  }
+
+  /// Errors that occur during lookup operation
+  variant lookup-error {
+    /// A completely unexpected error
+    unexpected(string),
+  }
+
+  /// Options that control the lookup operation
+  record lookup-options {
+    /// Timeout that should be used, in nanoseconds
+    /// If not specified, the implementer *may* provide a default timeout.
+    timeout-ns: option<u64>,
+
+    /// How and whether to retry the operation
+    retry-strategy: option<retry-strategy>,
+
+    /// A known span to associate this lookup with
+    parent-span: option<request-span>,
+  }
+
+  /// Errors that occur during lookup for an individual subdocument path
+  variant subdocument-lookup-error {
+    /// The provided path does not exist in the document
+    path-does-not-exist(string),
+    /// Type Conflict between the path in the document and path command
+    path-mismatch(string),
+    /// A document error
+    document-error(document-error),
+    /// A completely unexpected error
+    unexpected(string),
+  }
+
+  /// Perform a lookup of one or more specs in a given document
+  lookup: func(
+    id: document-id,
+    operations: list<lookup-operation>,
+    options: lookup-options,
+  ) -> result<list<result<document, subdocument-lookup-error>>, lookup-error>;
+
+  /// Perform a lookup on a particular document from any replica server
+  lookup-in-any-replicas: func(
+    id: document-id,
+    operations: list<lookup-operation>,
+    options: lookup-options,
+  ) -> result<list<result<document, subdocument-lookup-error>>, lookup-error>;
+
+  /// Perform a lookup on a particular document from all replica servers
+  lookup-in-all-replicas: func(
+    id: document-id,
+    operations: list<lookup-operation>,
+    options: lookup-options,
+  ) -> result<list<result<document, subdocument-lookup-error>>, lookup-error>;
```

Comment:
It's a list of [`LookupInReplicaResult`](https://pkg.go.dev/github.com/couchbase/gocb/v2#LookupInReplicaResult).

Reply:
I *think* this is the same concern as https://github.com/couchbaselabs/wasmcloud-provider-couchbase/pull/2#discussion_r1658458789 ? Is that right?

Reply:
Not exactly, because the function returns all copies of the requested sub-doc values. So it's a list of object that themselves contain a list of values.

Reply:
Ah, OK I see what you mean here, it's actually a list of the list of the changes, (per-replica)!

- [ ] fix the all-replicas case

Reply:
Done this should be fixed now! it's both a list of list and I fixed what the function was returning -- including the missing `cas` information